ts/pubsub.js

```diff
@@ -393,23 +406,22 @@ class BasicPubSub extends Pubsub {
 
     const from = this.peerId.toB58String()
 
-    const buildMessage = (msg, cb) => {
-      const seqno = utils.randomSeqno()
-      const msgObj = {
+    const buildMessage = data => {
+      return {
         from: from,
-        data: msg,
-        seqno: seqno,
+        data: data,
+        seqno: utils.randomSeqno(),
         topicIDs: topics
       }
-      // Emit to self if I'm interested and emitSelf enabled
-      this._options.emitSelf && this._emitMessages(topics, [msgObj])
-
-      return this._buildMessage(msgObj)
     }
-    const msgObjects = await pMap(messages, buildMessage)
+    const msgObjects = messages.map(buildMessage)
 
+    // Emit to self if I'm interested and emitSelf enabled
+    this._options.emitSelf && msgObjects.forEach(msg => this._emitMessage(msg))
+
+    const signMessage = (msg, cb) => this._buildMessage(msg)
     // send to all the other peers
-    this._publish(utils.normalizeOutRpcMessages(msgObjects))
+    this._publish(await pMap(msgObjects, signMessage))
```

Comment:
This was another thing I forgot to mention in the description.
Changed the function signature to accept an `InMessage`. (gossipsub._publish was already accepting `InMessage`, and normalizing there, this was an undiagnosed type error)

Changed in this direction (rather than changing `gossipsub._publish`), related to https://github.com/libp2p/js-libp2p-pubsub/issues/50, I think it makes sense to operate on `InMessage` when working with "message" objects whenever possible.

Reply:
sounds good!